core/controllers/story_editor.py

```
@@ -251,7 +251,10 @@ def put(self, story_id: str) -> None:
             story_publication_action
             == topic_domain.STORY_PUBLICATION_ACTION_PUBLISH
         ):
-            topic_services.publish_story(topic_id, story_id, self.user_id)
+            try:
+                topic_services.publish_story(topic_id, story_id, self.user_id)
+            except utils.ValidationError as e:
+                raise self.InvalidInputException(e)
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)